frontend/ui/widget/pagebrowserwidget.lua

```diff
@@ -595,6 +641,17 @@ function PageBrowserWidget:update()
         -- extended_sep_pages = extended_sep_pages,
     }
     self.row[1] = row
+    -- NT: update layout
+    -- remove existing BookMapRow from layout
+    if #self.layout > self.nb_rows then
+        for i = self.nb_rows + 1, #self.layout do
+            self.layout[i] = nil
+        end
+    end
+    -- add new BookMapRow layout
+    for _, focus_row in ipairs(row.focus_layout) do
+        table.insert(self.layout, focus_row)
+    end
 
```

Comment:
@poire-z is this the bit you say you would prefer not to be loaded on touch devices?

Reply:
One of the bitS.
Yes, this and its brothers would do.
Rather create a `self.build_focus_layout = true/false` early, so if we ever need to change the condition, it will be in a single place.

Reply:
Well, I'm not re-reading the whole thing.
I'd like anything that build some stuff and `table.insert()` it into a `something_layout` to be wrapped by this `if self.build_focus_layout then`.

Also, I think the rendering/shifting issue with using the border to highlight the focused item - cf https://github.com/koreader/koreader/pull/12583#issuecomment-2427403108 - should be fixed.
ie FrameContainer could have a `draw_only_border` that would not interact with the sizing, but would just be drawn - outside or inside, I dunno, possibly overflowing on the inner content or outer neighbours, avoiding that being the job of the designer to provide some extra margin or padding.

Reply:
The VirtualKeyboard keys have a highlight frame that's drawn inside, as a point of reference (hell, that might even be a FrameContainer feature, can't recall ;p).

Reply:
that might be the (your) inner_border I mentionned yesterday :)

Reply:
I wish you could come back from retirement @comphilip and have a look at this.

Reply:
I believe the whole wrap business is done (I hope I didn't miss anything), could you look into that border thing @poire-z, I have no clue what's going on there... please. ;)

Reply:
I spent a few hours today working on adjusting the bookmapwidget from this PR, I fixed the border thing (you'll have to tell later if and how thiner we can go than 3*Size.border.thin).
Haven't yet looked at the PageBrowser widget.
So, good if you're done - don't add stuff to this PR so we don't conflict.

Reply:
understood.